src/types.ts

```diff
@@ -10,7 +10,7 @@ export type TCoreAttributes = {
     user_id?: string
     app_id?: string
     user_identity?: string
-} & TGrowthbookAttributes
+} & Partial<TGrowthbookAttributes>
```

Comment:
We even don't need `id` to be mandatory?

Reply:
not for this `setAttribute` function, because this is being used as a way to set attributes on both rudderstack and growthbook, we don't necessarily need to pass growthbook attributes since growthbook usage is optional, the required `id` is causing the type error once we try to set the `device_type` attribute.

One thing we could do is to add the `device_type` property into the rudderstack attributes type, that way I think it will not throw the error. let me check on that

Reply:
I tried that it didn't work, we need to keep the `GrowthbookAttributes` as optional

Reply:
ok thanks for clarification.